src/test/java/io/jenkins/infra/repository_permissions_updater/JiraAPITest.java

```diff
@@ -0,0 +1,223 @@
+package io.jenkins.infra.repository_permissions_updater;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import io.jenkins.infra.repository_permissions_updater.helper.HttpUrlStreamHandler;
+import io.jenkins.infra.repository_permissions_updater.helper.MemoryAppender;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLStreamHandlerFactory;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+import java.util.function.Supplier;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+class JiraAPITest {
+
+    private static MemoryAppender memoryAppender;
+    private static HttpUrlStreamHandler httpUrlStreamHandler;
+
+    private static final Supplier<String> JIRA_URL = () -> System.getProperty("jiraUrl", "https://issues.jenkins.io");
+    private static final  Supplier<String> JIRA_USE_URL = () -> JIRA_URL.get() + "/rest/api/2/user";
+    private static final  Supplier<String> JIRA_USER_QUERY = () -> JIRA_USE_URL.get() + "?username=%s";
+    private static final Supplier<String> JIRA_COMPONENTS_URL = () -> JIRA_URL.get() + "/rest/api/2/project/JENKINS/components";
```

Comment:
Why are these using suppliers rather than just static strings?

Reply:
Otherwise many tests would fail because the URLs would then only be created once and thus errors would occur in the different test cases

Reply:
can you delete the test cases I asked about in https://github.com/jenkins-infra/repository-permissions-updater/pull/3936#pullrequestreview-2113190695 and then potentially you won't need this.

Otherwise you could use dependency injection instead by adding the url in the constructor which would be cleaner